File: src/app/shared/model/livro.ts
import { Exemplar } from "./exemplar";

export class Livro {

    id: number; 
    titulo: string; 
    autor: string; 
    descricao: string; 
    exemplares: Exemplar[];


    constructor(id: number, titulo: string, autor: string, descricao: string) {
        this.id=id;
        this.titulo=titulo;
        this.autor=autor;
        this.descricao=descricao;
        this.exemplares=[];
    }

    /** 
    public getId(): number{
        return this.id;
    }
    public getTitulo(): string{
        return this.titulo;
    }
    public getAutor(): string{
        return this.autor;
    }
    public getDescricao(): string{
        return this.descricao;
    }

    public setId(id: number){
        this.id=id;
    }

    public setTitulo(titulo: string){
        this.titulo=titulo;
    }

    public setAutor(autor: string){
        this.autor=autor;
    }

    public setDescricao(descricao: string){
        this.descricao=descricao;
    }
    */
}

File: demo/src/main/java/com/livro/demo/model/Livro.java
package com.livro.demo.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "tb_livro")
public class Livro {

   @Id
   @GeneratedValue(strategy = GenerationType.AUTO)
   private Integer id;
   private String titulo;
   private String autor;
   private String descricao;

   public Integer getId() {
       return id;
   }

   public String getTitulo() {
       return titulo;
   }

   public void setTitulo(String titulo) {
       this.titulo = titulo;
   }

   public String getAutor() {
       return autor;
   }

   public void setAutor(String autor) {
       this.autor = autor;
   }

   public String getDescricao() {
       return descricao;
   }

   public void setDescricao(String descricao) {
       this.descricao = descricao;
   }
}
